packages/ui-keyring/src/options/index.ts

```diff
@@ -129,10 +133,25 @@ class KeyringOption implements KeyringOptionInstance {
       });
   }
 
+  private addContracts (keyring: KeyringStruct, options: KeyringOptions): void {
+    const available = keyring.contracts.subject.getValue();
+
+    Object
+      .values(available)
+      .filter(({ json }: SingleAddress) => {
+        return !json.meta.isRecent;
+      })
+      .sort(sortByName)
+      .forEach(({ option }: SingleAddress) => {
+        options.contract.push(option);
+      });
+  }
+
   private emptyOptions (): KeyringOptions {
     return {
       account: [],
       address: [],
+      contract: [],
```

Comment:
Not 100% sure we want to include contracts in here - since it will appear in all Address dropdowns?